### PR TITLE
dist lookup table only support sgd

### PR DESCRIPTION
**Pull request by jacquesqiao**
1. dist lookup table only supports SGD while other dense tables support all optimizer.
2. add some unit test for sync distributed lookup table

### CODE REVIEW

File: python/paddle/fluid/transpiler/distribute_transpiler.py [hunk 1]
```diff
@@ -937,11 +935,8 @@ def _create_table_optimize_block(self, pserver_index, pserver_program,
             "LearningRate": [lr_var]
         }
         outputs = {"ParamOut": [param_var]}
-        table_opt_block.append_op(
-            type=table_opt_op.type,
-            inputs=inputs,
-            outputs=outputs,
-            attrs=table_opt_op.attrs)
+        # only support sgd now
+        table_opt_block.append_op(type="sgd", inputs=inputs, outputs=outputs)
```
Comment (typhoonzero) [inline python/paddle/fluid/transpiler/distribute_transpiler.py]: Could print a warning here to indicate users.

File: python/paddle/fluid/tests/unittests/test_dist_transpiler.py [hunk 2]
```diff
@@ -359,5 +359,111 @@ def transpiler_test_impl(self):
...
+
+class TestLocalLookupTable(TestDistLookupTableBase):
+    def net_conf(self):
+        self.network_with_table(is_sparse=True, is_distributed=False)
+
+    def transpiler_test_impl(self):
+        pserver1, startup1 = self.get_pserver(self.pserver1_ep)
+
+        self.assertEqual(len(pserver1.blocks), 3)
+        # print(str(pserver1))
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_dist_transpiler.py]: Can remove this line.

Comment (jacquesqiao) [reply python/paddle/fluid/tests/unittests/test_dist_transpiler.py]: done

File: python/paddle/fluid/transpiler/distribute_transpiler.py (hunk 1 above)
Comment (jacquesqiao) [reply python/paddle/fluid/transpiler/distribute_transpiler.py]: done

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**